gen2-pointcloud/save_point_cloud/main.py

```diff
@@ -0,0 +1,232 @@
+queueNames.append("rgb")
+depthOut.setStreamName("depth")
+queueNames.append("depth")
+
+# Properties
+camRgb.setBoardSocket(dai.CameraBoardSocket.RGB)
+camRgb.setResolution(dai.ColorCameraProperties.SensorResolution.THE_1080_P)
+camRgb.setFps(fps)
+if downscaleColor:
+    camRgb.setIspScale(2, 3)
+# For now, RGB needs fixed focus to properly align with depth.
+# This value was used during calibration
+try:
+    calibData = device.readCalibration2()
+    lensPosition = calibData.getLensPosition(dai.CameraBoardSocket.RGB)
+    if lensPosition:
+        camRgb.initialControl.setManualFocus(lensPosition)
+except:
+    raise
+    
+left.setResolution(monoResolution)
+left.setBoardSocket(dai.CameraBoardSocket.LEFT)
+left.setFps(fps)
+right.setResolution(monoResolution)
+right.setBoardSocket(dai.CameraBoardSocket.RIGHT)
+right.setFps(fps)
+
+stereo.setDefaultProfilePreset(dai.node.StereoDepth.PresetMode.HIGH_DENSITY)
+# LR-check is required for depth alignment
+stereo.setLeftRightCheck(True)
+stereo.setDepthAlign(dai.CameraBoardSocket.RGB)
+
+# Linking
+camRgb.isp.link(rgbOut.input)
+left.out.link(stereo.left)
+right.out.link(stereo.right)
+stereo.depth.link(depthOut.input)
+# rectified stereo
+xoutRectifLeft = pipeline.create(dai.node.XLinkOut)
+xoutRectifRight = pipeline.create(dai.node.XLinkOut)
+xoutRectifLeft.setStreamName("rectifiedLeft")
+xoutRectifRight.setStreamName("rectifiedRight")
+stereo.rectifiedLeft.link(xoutRectifLeft.input)
+stereo.rectifiedRight.link(xoutRectifRight.input)
+
+# Connect to device and start pipeline
+with device:
+    device.setIrLaserDotProjectorBrightness(1200)
+    serial_number = device.getMxId()
+    timestamp = str(int(time.time()))
+    calibFile = f"{serial_number}_{timestamp}_calibration.json"
+    calibData = device.readCalibration()
+    calibData.eepromToJsonFile(calibFile)
+    print("wrote", calibFile)
+
+    M_rgb = np.array(
+        calibData.getCameraIntrinsics(dai.CameraBoardSocket.RGB, width, height)
+    )
+
+    device.startPipeline(pipeline)
+
+    frameRgb = None
+    frameDepth = None
+
+    # Configure windows; trackbar adjusts blending ratio of rgb/depth
+    rgbWindowName = "rgb"
+    depthWindowName = "depth"
+    blendedWindowName = "rgb-depth"
+    cv2.namedWindow(rgbWindowName)
+    cv2.namedWindow(depthWindowName)
+    cv2.namedWindow(blendedWindowName)
+    cv2.createTrackbar(
+        "RGB Weight %", blendedWindowName, int(rgbWeight * 100), 100, updateBlendWeights
+    )
+
+    while True:
+        latestPacket = {
+            "rgb": None,
+            "depth": None,
+            "rectifiedLeft": None,
+            "rectifiedRight": None,
+        }
+
+        queueEvents = device.getQueueEvents(
+            ("rgb", "depth", "rectifiedLeft", "rectifiedRight")
+        )
+        for queueName in queueEvents:
+            packets = device.getOutputQueue(queueName).tryGetAll()
+            if len(packets) > 0:
+                latestPacket[queueName] = packets[-1]
+
```

Comment:
This logic of getting frames isn't ideal. Maybe jut use get() instead, or maybe add syncing logic to it (based on sequence number)?